core/translate/plan.rs

```diff
@@ -0,0 +1,469 @@
+use core::fmt;
+use std::{
+    fmt::{Display, Formatter},
+    rc::Rc,
+};
+
+use sqlite3_parser::ast;
+
+use crate::{function::AggFunc, schema::BTreeTable, util::normalize_ident, Result};
+
+pub struct Plan {
+    pub root_node: Operator,
+    pub referenced_tables: Vec<(Rc<BTreeTable>, String)>,
+}
+
+impl Display for Plan {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.root_node)
+    }
+}
+
+/**
+  An Operator is a Node in the query plan.
+  Operators form a tree structure, with each having zero or more children.
+  For example, a query like `SELECT t1.foo FROM t1 ORDER BY t1.foo LIMIT 1` would have the following structure:
+    Limit
+      Order
+        Project
+          Scan
+*/
+#[derive(Clone, Debug)]
+pub enum Operator {
```

Comment:
Any reason to not call this `Node` or `PlanNode`? Since you already explain in the comment as it being a node, it made sense in my head.

Reply:
I would comments for each Operator too.

Reply:
That was the original name and I renamed it after Pekka's comment 😂 

Maybe the name doesn't really matter? :)

Reply:
I actually went the opposite way and finished renaming all the leftover "node" variables and types to "operator" so it's at least consistent

Reply:
> That was the original name and I renamed it after Pekka's comment 😂
> 
> Maybe the name doesn't really matter? :)

Oh oops, naming is hard :P